### PR TITLE
Purge legacy atomics (in favor of desul atomics)

### PR DESCRIPTION
**Pull request by dalg24**
Step 2 in #5789

Macro `KOKKOS_ENABLE_IMPL_DESUL_ATOMICS` was removed too

### CODE REVIEW

**Comment by dalg24** [comment]
(had messed up HIP backend)

File: core/src/HIP/Kokkos_HIP_Locks.hpp [hunk 1]
```diff
@@ -138,20 +135,6 @@ inline static void ensure_hip_lock_arrays_on_device() {
...
 
-#else
-
-#ifdef KOKKOS_ENABLE_HIP_RELOCATABLE_DEVICE_CODE
-inline void ensure_hip_lock_arrays_on_device() {}
-#else
-// Still Need copy_hip_lock_arrays for team scratch etc.
-inline static void ensure_hip_lock_arrays_on_device() {
-  copy_hip_lock_arrays_to_device();
-  desul::ensure_hip_lock_arrays_on_device();
```
Comment (masterleinad) [inline core/src/HIP/Kokkos_HIP_Locks.hpp:149]: This looks like you deleted the wrong branch. We should keep the one that has
```
desul::ensure_hip_lock_arrays_on_device();
```

**Review by masterleinad — CHANGES_REQUESTED** [review]

File: core/src/HIP/Kokkos_HIP_Locks.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/HIP/Kokkos_HIP_Locks.hpp:149]: Yup you are right

Comment (dalg24) [reply core/src/HIP/Kokkos_HIP_Locks.hpp]: Fixed

**Review by masterleinad — APPROVED** [review]

**Review by Rombur — APPROVED** [review]

**Comment by crtrott** [comment]
Do you want to remove the classic atomic lock array later?

**Comment by dalg24** [comment]
> Do you want to remove the classic atomic lock array later?

Yes.

**Review by crtrott — APPROVED** [review]

**Pull request merged**